client/client-multi/client-c/src/kaa_event.c

```diff
@@ -108,6 +120,58 @@ extern kaa_sync_handler_fn kaa_channel_manager_get_sync_handler(kaa_channel_mana
                                                               , kaa_service_t service_type);
 
 
+static void destroy_event_listener_request(void *request_p)
+{
+    if (request_p) {
```

Comment:
To reduce the amount of nesting levels you can use
KAA_RETURN_IF_NIL(request_p,);